esrally/driver/runner.py

```diff
@@ -245,6 +255,18 @@ def __call__(self, es, params):
             raise exceptions.DataError(
                 "Bulk parameter source did not provide a 'bulk-size' parameter. Please add it to your parameter source.")
 
+        bulk_request_size_bytes = 0
+        total_document_size_bytes = 0
+
+        for i in range(len(params["body"])):
```

Comment:
This adds some overhead which is - depending on the track it's - at worst in the single-digit percentage range (I've measured up to 95ms). I think it would make sense to move this loop to the method `detailed_stats`. Users can then enable it for their tracks by setting `detailed-results` to `true. Complete example:

```json
{
  "name": "bulk-index",
  "operation-type": "index",
  "bulk-size": 5000,
  "detailed-results": true
}
```

Btw, you could also use [`enumerate`](https://docs.python.org/3.6/library/functions.html#enumerate) for the loop instead of range to simplify it a bit:

```python
for line_number, data in enumerate(params["body"]):
    line_size = len(data)
    if params["action_metadata_present"]:
        total_document_size_bytes += line_size
    #....
```

Another fine detail: the `len` function in Python returns the number of characters (although it's not really clearly documented) but combining the following two facts show that this is the case:

1. "[len] [r]eturn[s] the length (the number of items) of an object. The argument may be a sequence (such as a string [...])" ([source](https://docs.python.org/3.6/library/functions.html#len))
2. "Strings are immutable sequences of Unicode code points." ([source](https://docs.python.org/3.6/library/stdtypes.html#textseq))

Therefore, we can reason that `len` returns the number of Unicode code points (i.e. characters) of a string. For characters that can be represented with 1 byte, the number of characters is identical to the number of bytes but strictly speaking it's the number of characters. To determine the number of bytes you need to encode the string as UTF-8, e.g. `len(s.encode('utf-8'))`.